### ISSUE TITLE
[FEATURE] Update progress bar total

### ISSUE THREAD

**Issue by kwilinsi** [sjmikler/progress-table#35]
Maybe I'm missing something, but is there no way to update the total of a progress bar? Why is `TableProgressBar._total` protected? I've tried changing it and haven't run into any obvious problems.

I'm writing a program that scans through files/directories and displays some summary statistics. I want a total progress bar at the bottom to estimate how long it will take. I don't know the total number of files from the outset, so I can't set `_total` right away. But I'm able to make an educated guess and refine that estimate over time.

Looks like in the [download_v2](https://github.com/sjmikler/progress-table/blob/main/examples/download_v2.py) example, you circumvent this by setting the total to `1` and then using `reset()` to set a proportion for each download. That could work, but it's rather clunky, especially if I want to show throughput and progress.

**Comment by szmikler** [comment]
Hey, thanks for creating the issue.

You're totally right. I wanted to make the PBAR methods similar to those available in TQDM but I misunderstood how `reset` function works in TQDM. Of course, the **reset** function **resets** the progress bar to 0; optionally setting up a new `total` value. My implementation doesn't work like this.

I want to fix the `reset` function, so you would be able to easily set a new `total`. The new signature would be the following:

```
def reset(total=None)
    ...
```
```
Resets to 0 iterations for repeated use.

Parameters
total: int or float, optional. Total to use for the new bar.
```

Let me know if that will work for you.

**Comment by szmikler** [comment]
I will be adding more functions, so it might be more convenient:

```

    def update(self, n=1):
        """Update the progress bar steps.

        Args:
            n: Number of steps to update the progress bar.
        """
        self._step += n

    def reset(self, total=None):
        """Reset the progress bar.

        Args:
            total: Modify the total number of iterations. Optional.
        """
        self._step = 0

        if total:
            self._total = total

    def set_step(self, step):
        """Overwrite the current step.

        Args:
            step: New value of the current step.
        """
        self._step = step

    def set_total(self, total):
        """Overwrite the total number of iterations.

        Args:
            total: New value of the total number of iterations
        """
        self._total = total
```

**Comment by szmikler** [comment]
I implemented changes described above in version 2.3.0. For your use case, you probably want to use `set_total` function.
